sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -246,7 +248,23 @@ namespace Azure { namespace Core { namespace Test {
         }
         if (clientSecret.empty())
         {
-          m_testCredential = std::make_shared<Azure::Identity::DefaultAzureCredential>();
+          try
+          {
+            // the ENVs are defined only by the pipeline and not by the user thus this will throw
```

Comment:
we cannot use chained token creds as those envs are defined only in the pipeline

Reply:
Is the bug because of GetEnv("") throwing or because we pass null strings to the constructor? Either way, AzurePipelineCredential is broken if it can't be used in the ChainedTokenCredential. Please open a bug for this.

Add a comment here that this is a workaround due to that bug. We should remove the workaround and use a ChainedTokenCredential.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/5769